universe/base.go

```diff
@@ -249,6 +250,49 @@ func (a *MintingArchive) verifyIssuanceProof(ctx context.Context,
 	return assetSnapshot, nil
 }
 
+// RegisterNewIssuanceBatch inserts a batch of new minting leaves within the
+// target universe tree (based on the ID), stored at the base key(s). We assume
+// the proofs within the batch have already been checked that they don't yet
+// exist in the local database.
+func (a *MintingArchive) RegisterNewIssuanceBatch(ctx context.Context,
+	items []*IssuanceItem) error {
+
+	log.Infof("Verifying %d new proofs for insertion into Universe",
+		len(items))
+
+	for idx := range items {
+		assetSnapshot, err := a.verifyIssuanceProof(ctx, items[idx])
```

Comment:
Can verify them in parallel (using a similar `Collect` pattern) here.

Reply:
Yeah, seems simple enough. Though on my machine this only took around 105ms for 100 proofs with the old code anyway. But this seems to have brought it down to around 13ms per 100 proofs :exploding_head: